File: Data Structures & Algorithms/duplicate-integer/submission-3.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    // hasDuplicate(nums) {
    //     if(!nums || nums.length === 0) {
    //         return false;
    //     };

    //     const numObj = {};

    //     for(let i = 0; i < nums.length; i++){
    //         if(numObj[nums[i]]) {
    //             return true;
    //         } else {
    //             numObj[nums[i]] = true;
    //         }
    //     }

    //     return false;
    // }
    hasDuplicate(nums) {
        const numsSet = new Set();
        for (const num of nums) {
            if(numsSet.has(num)) return true;

            numsSet.add(num);
        }

        return false;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums_set = set(nums)
        max_consecutive = 0
        for n in nums_set:
            consecutive = 1
            while n + consecutive in nums_set:
                consecutive += 1
            max_consecutive = max(max_consecutive, consecutive)

        return max_consecutive